[Bug]: Failed to build package 'coreutils'

### Problem description

Since the NDK was bumped to version 27, compiling `coreutils` gives this error

```
...
  CC       lib/libcoreutils_a-nstrftime.o
  CC       lib/libcoreutils_a-obstack.o
  CC       lib/libcoreutils_a-openat-die.o
  CC       lib/libcoreutils_a-openat-safer.o
In file included from /home/builder/.termux-build/coreutils/src/lib/nstrftime.c:19:
/home/builder/.termux-build/coreutils/src/lib/strftime.c:1725:17: error: call to undeclared function 'mktime_z'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 1725 |             t = mktime_z (tz, &ltm);
      |                 ^
/home/builder/.termux-build/coreutils/src/lib/strftime.c:1725:17: note: did you mean 'mktime'?
/home/builder/.termux-build/_cache/android-r27-api-24-v0/bin/../sysroot/usr/include/time.h:159:8: note: 'mktime' declared here
  159 | time_t mktime(struct tm* _Nonnull __tm);
      |        ^
1 error generated.
make[2]: *** [Makefile:16274: lib/libcoreutils_a-nstrftime.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/builder/.termux-build/coreutils/build'
make[1]: *** [Makefile:22447: all-recursive] Error 1
make[1]: Leaving directory '/home/builder/.termux-build/coreutils/build'
make: *** [Makefile:8865: all] Error 2
Failed to build package 'coreutils' for arch 'aarch64'
[*] Building 'coreutils' exited with exit code 2
Error: Process completed with exit code 2.
```

### What steps will reproduce the bug?

`coreutils` should be in `PACKAGES` in `build-bootstraps.sh`:

```
./scripts/run-docker.sh ./clean.sh
./scripts/run-docker.sh ./scripts/build-bootstraps.sh --android10
```

### What is the expected behavior?

_No response_

### System information

```shell
N/A
```

@Biswa96 Then I get

```
  CC       src/libsinglebin_cut_a-cut.o
  CC       src/libsinglebin_date_a-date.o
  CC       src/libsinglebin_dd_a-dd.o
  CC       src/libsinglebin_dir_a-coreutils-dir.o
/home/builder/.termux-build/coreutils/src/src/date.c:578:14: error: incompatible integer to pointer conversion initializing 'timezone_t' (aka 'struct __timezone_t *') with an expression of type 'int' [-Wint-conversion]
  578 |   timezo  CC       src/libsinglebin_ls_a-ls.o
ne_t tz = tzalloc (tzstring);
      |              ^    ~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [Makefile:20152: src/libsinglebin_date_a-date.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/home/builder/.termux-build/coreutils/src/src/ls.c:1727:11: error: incompatible integer to pointer conversion assigning to 'timezone_t' (aka 'struct __timezone_t *') from 'int' [-Wint-conversion]
 1727 |   localtz = tzalloc (getenv ("TZ"));
      |           ^ ~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [Makefile:20698: src/libsinglebin_ls_a-ls.o] Error 1
make[2]: Leaving directory '/home/builder/.termux-build/coreutils/build'
make[1]: *** [Makefile:22447: all-recursive] Error 1
make[1]: Leaving directory '/home/builder/.termux-build/coreutils/build'
make: *** [Makefile:8865: all] Error 2
Failed to build package 'coreutils' for arch 'arm'
[*] Building 'coreutils' exited with exit code 2
Error: Process completed with exit code 2.
```

@licy183 If it is any helpful, this is the output when setting `ac_cv_type_timezone_t=no`:

```
  CC       lib/libcoreutils_a-long-options.o
  CC       lib/libcoreutils_a-malloca.o
  CC       lib/libcoreutils_a-math.o
  CC       lib/libcoreutils_a-mbrlen.o
In file included from /home/builder/.termux-build/coreutils/src/lib/malloca.cIn file included from /home/builder/.termux-build/coreutils/src/lib/long-options.c:27:
In file included from ./lib/stdio.h:95:
In file included from ./lib/sys/stat.h:51:
./lib/time.h:998:25: error: typedef redefinition with different types ('struct tm_zone *' vs 'struct __timezone_t *')
  998 | typedef struct tm_zone *timezone_t;
      |                         ^
/home/builder/.termux-build/_cache/android-r27-api-24-v0/bin/../sysroot/usr/include/time.h:52:30: note: previous definition is here
   52 | typedef struct __timezone_t* timezone_t;
      |                              ^
:22:
In file included from /home/builder/.termux-build/coreutils/src/lib/malloca.h:29:
In file included from ./lib/stdlib.h:36:
In file included from /home/builder/.termux-build/_cache/android-r27-api-24-v0/bin/../sysroot/usr/include/stdlib.h:34:
In file included from /home/builder/.termux-build/_cache/android-r27-api-24-v0/bin/../sysroot/usr/include/malloc.h:30:
In file included from ./lib/stdio.h:95:
In file included from ./lib/sys/stat.h:51:
./lib/time.h:998:25: error: typedef redefinition with different types ('struct tm_zone *' vs 'struct __timezone_t *')
  998 | typedef struct tm_zone *timezone_t;
      |                         ^
/home/builder/.termux-build/_cache/android-r27-api-24-v0/bin/../sysroot/usr/include/time.h:52:30: note: previous definition is here
   52 | typedef struct __time1 error generated.
zone_t* timezone_t;
      |                              ^
make[2]: *** [Makefile:15616: lib/libcoreutils_a-long-options.o] Error 1
make[2]: *** Waiting for unfinished jobs....
1 error generated.
In file included from /home/builder/.termux-build/coreutils/src/lib/mbrlen.c:21:
In file included from ./lib/wchar.h:80:
In file included from /home/builder/.termux-build/_cache/android-r27-api-24-v0/bin/../sysroot/usr/include/wchar.h:33:
In file included from ./lib/stdio.h:95:
In file included from ./lib/sys/stat.h:51:
./lib/time.h:998:25: error: typedef redefinition with different types ('struct tm_zone *' vs 'struct __timezone_t *')
  998 | typedef struct tm_zone *timezone_t;
      |                         ^
/home/builder/.termux-build/_cache/android-r27-api-24-v0/bin/../sysroot/usr/include/time.h:52:30: note: previous definition is here
   52 | typedef struct __timezone_t* timezone_t;
      |                              ^
make[2]: *** [Makefile:15658: lib/libcoreutils_a-malloca.o] Error 1
1 error generated.
make[2]: *** [Makefile:15686: lib/libcoreutils_a-mbrlen.o] Error 1
make[3]: Leaving directory '/home/builder/.termux-build/coreutils/build'
make[2]: Leaving directory '/home/builder/.termux-build/coreutils/build'
make[1]: *** [Makefile:22447: all-recursive] Error 1
make[1]: Leaving directory '/home/builder/.termux-build/coreutils/build'
make: *** [Makefile:8865: all] Error 2
Failed to build package 'coreutils' for arch 'aarch64'
[*] Building 'coreutils' exited with exit code 2
Error: Process completed with exit code 2.
```

--- a/ndk-patches/27/time.h.patch
+++ b/ndk-patches/27/time.h.patch
@@ -1,0 +1,18 @@
+--- a/usr/include/time.h
++++ b/usr/include/time.h
+@@ -39,6 +39,7 @@
+ 
+ __BEGIN_DECLS
+ 
++#if __ANDROID_API__ >= 35
+ /* If we just use void* in the typedef, the compiler exposes that in error messages. */
+ struct __timezone_t;
+ 
+@@ -50,6 +51,7 @@
+  * to remove the pointer.
+  */
+ typedef struct __timezone_t* timezone_t;
++#endif
+ 
+ /** Divisor to compute seconds from the result of a call to clock(). */
+ #define CLOCKS_PER_SEC 1000000

--- a/packages/coreutils/build.sh
+++ b/packages/coreutils/build.sh
@@ -3,6 +3,7 @@ TERMUX_PKG_DESCRIPTION="Basic file, shell and text manipulation utilities from t
 TERMUX_PKG_LICENSE="GPL-3.0"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION=9.5
+TERMUX_PKG_REVISION=1
 TERMUX_PKG_SRCURL=https://mirrors.kernel.org/gnu/coreutils/coreutils-${TERMUX_PKG_VERSION}.tar.xz
 TERMUX_PKG_SHA256=cd328edeac92f6a665de9f323c93b712af1858bc2e0d88f3f7100469470a1b8a
 TERMUX_PKG_DEPENDS="libandroid-support, libgmp, libiconv"

--- a/scripts/build/termux_step_setup_toolchain.sh
+++ b/scripts/build/termux_step_setup_toolchain.sh
@@ -7,7 +7,7 @@ termux_step_setup_toolchain() {
 		# toolchain setup to ensure that everyone gets an updated
 		# toolchain
 		if [ "${TERMUX_NDK_VERSION}" = "27" ]; then
-			TERMUX_STANDALONE_TOOLCHAIN+="-v0"
+			TERMUX_STANDALONE_TOOLCHAIN+="-v1"
 			termux_setup_toolchain_27
 		elif [ "${TERMUX_NDK_VERSION}" = 23c ]; then
 			TERMUX_STANDALONE_TOOLCHAIN+="-v8"

